supervision/keypoint/core.py

```diff
@@ -136,6 +136,48 @@ def from_ultralytics(cls, ultralytics_results) -> KeyPoints:
         data = {CLASS_NAME_DATA_FIELD: class_names}
         return cls(xy, class_id, confidence, data)
 
+    @classmethod
+    def from_yolo_nas(cls, yolo_nas_results) -> KeyPoints:
+        """
+        Create a KeyPoints instance from a YOLO NAS results.
+
+        Args:
+            yolo_nas_results (ImagePoseEstimationPrediction):
+                The output object from YOLO NAS.
+
+        Returns:
+            KeyPoints: A new KeyPoints object.
+
+        Example:
+            ```python
+            import cv2
+            import supervision as sv
+            import super_gradients
+
+            image = cv2.imread(<SOURCE_IMAGE_PATH>)
+
+            yolo_nas = super_gradients.training.models.get(
+                "yolo_nas_pose_s", pretrained_weights="coco_pose").to("cuda")
```

Comment:
Using `to("cuda")` seems quite risky. It will crash if someone copies the snippet in an environment without a GPU. I would add `"cuda" if torch.cuda.is_available() else "cpu"` to the example. Just to be safe.